### ISSUE TITLE
enhance curly option

### ISSUE THREAD

**Issue by Imarok** [eslint/eslint#8957]
The curly rule should have a new option that works like multi, but enforces braces around if-else and switch-case:

**What rule do you want to change?**
curly
**Does this change cause the rule to produce more or fewer warnings?**
depends on
**How will the change be implemented? (New option, new default behavior, etc.)?**
new option
**Please provide some example code that this change will affect:**
This should be correct code with the new option:
```js
	while(foo)
	{ // bad with multi
		if(bar)
			warn(sd);
		else
			warn(sd);
	} // bad with multi

	if(foo)
	{ // bad with multi
		switch (n)
		{
		case 10:
			return 1;

		case 20:
			foo();
		}
	} // bad with multi
```
This code is correct with the option `multi` and should also be correct with the new option:
```js
	while(foo)
		for(let bar; bar > 0; bar++)
			if(foobar)
				dosomething();
```

**What does the rule currently do for this code?**
See behaviour with `curly: ["warn", "multi"]` above (// bad with multi)

**What will the rule do after it's changed?**
behave like the multi option but enforce curlies around if-else and switch-case. (see above)

This is needed, because omitting the curly for simple statements improves readability and reduces LOC, but for things like if-else or switch-case, the curly is needed for readability.

**Comment by ilyavolodin** [comment]
I agree. I think this is a personal preference and it's not a good fit for the core. Difference between `if..else` and `while` is not obvious to me.

**Comment by Imarok** [comment]
> I agree. I think this is a personal preference and it's not a good fit for the core. Difference between if..else and while is not obvious to me.

`while` has only one scope:
```js
while(foo)
{
  scope1
}
```
`if..else` has two scopes:
```js
if
{
  scope1
}
else
{
  scope2
}
```

So at the first glance without looking at what specifically the keyword is (`while`, `if` or `else`) you cannot say if the second scopes still belongs to the thing above or not.

**Comment by not-an-aardvark** [comment]
Closing because this has three :-1:s from the team, so it seems like the team is not in favor of adding this to core. However, you can always create a custom rule for this and distribute it as part of a plugin.
